internal/datastore/spanner/spanner.go

```diff
@@ -12,6 +12,7 @@ import (
 	"cloud.google.com/go/spanner"
 	ocprom "contrib.go.opencensus.io/exporter/prometheus"
 	sq "github.com/Masterminds/squirrel"
+	"github.com/google/uuid"
```

Comment:
same suggestion as with CRDB: we can have Spanner generate the UUID  value and spare the network transfer, using `DEFAULT (GENERATE_UUID()),`. See https://cloud.google.com/spanner/docs/schema-design#uuid_primary_key

EDIT: I see this won't work here because we need to provide the transaction tag in the Spanner protocol 👍🏻

Reply:
Yeah, unfortunately